dropboxfs.py

```diff
@@ -183,152 +265,160 @@ def metadata(self, path, cache_read=True):
         try:
-            metadata = super(DropboxClient, self).file_move(src, dst)
-        except rest.ErrorResponse, e:
-            if e.status == 404:
+            metadata = super(DropboxClient, self).files_move(src, dst)
+        except ApiError, e:
+            if e.error.is_from_lookup() and e.error.get_from_lookup().is_not_found():
                 raise ResourceNotFoundError(src)
-            if e.status == 403:
+            if e.error.is_to() and e.error.get_to().is_conflict():
                 raise DestinationExistsError(dst)
             LOGGER.error(e, exc_info=True, extra={'stack': True,})
-            raise RemoteConnectionError(opname='file_move', path=path,
+            raise RemoteConnectionError(opname='file_move', path=src,
                                         details=e)
         self.cache.pop(src, None)
         self.cache.set(dst, metadata)
 
-    def file_delete(self, path):
+    def files_delete(self, path):
         try:
-            super(DropboxClient, self).file_delete(path)
-        except rest.ErrorResponse, e:
-            if e.status == 404:
+            super(DropboxClient, self).files_delete(path)
+        except ApiError, e:
+            if e.error.is_path_lookup() and e.error.get_path_lookup().is_not_found():
                 raise ResourceNotFoundError(path)
-            if e.status == 400 and 'must not be empty' in str(e):
-                raise DirectoryNotEmptyError(path)
-            raise
+            raise RemoteConnectionError(opname='file_delete', path=path,
+                                        details=e)
         self.cache.pop(path, None)
 
-    def put_file(self, path, f, overwrite=False):
+    def files_upload(self, f, path, mode=WriteMode('add', None)):
         try:
-            super(DropboxClient, self).put_file(path, f, overwrite=overwrite)
-        except rest.ErrorResponse, e:
+            super(DropboxClient, self).files_upload(f, path, mode)
+        except ApiError, e:
             LOGGER.error(e, exc_info=True, extra={'stack': True,})
             raise RemoteConnectionError(opname='put_file', path=path,
                                         details=e)
         self.cache.pop(dirname(path), None)
 
 
-def create_client(app_key, app_secret, access_type, token_key, token_secret):
-    """Uses token from create_token() to gain access to the API."""
-    s = session.DropboxSession(app_key, app_secret, access_type)
-    s.set_token(token_key, token_secret)
-    return DropboxClient(s)
+def create_client(token):
+    """Uses token to gain access to the API."""
+    return DropboxClient(token)
 
 
 def metadata_to_info(metadata, localtime=False):
-    isdir = metadata.pop('is_dir', False)
+    isdir = isinstance(metadata, FolderMetadata)
     info = {
-        'size': metadata.pop('bytes', 0),
+        'size': getattr(metadata, 'size', 0),
         'isdir': isdir,
         'isfile': not isdir,
+        'modified_time': getattr(metadata, 'server_modified', None),
```

Comment:
It may be good to do a timezone conversion to match Indiana since that is the default behavior of `SmartFileFS`. I think this would be good for consistency.